dmoj/executors/RUST.py

```diff
@@ -62,9 +62,14 @@ def get_versionable_commands(cls):
 
     def get_compile_args(self):
         args = [self.get_command(), 'build', '--release']
+        if bool_env('DMOJ_CARGO_SHARED_TARGET'):
+            args += ['--target-dir', os.path.expanduser('~/.cargo/dmoj-shared-target')]
         if bool_env('DMOJ_CARGO_OFFLINE'):
             args += ['--offline']
         return args
 
     def get_compiled_file(self):
-        return self._file('target', 'release', 'user_submission')
+        if bool_env('DMOJ_CARGO_SHARED_TARGET'):
+            return os.path.expanduser('~/.cargo/dmoj-shared-target/release/user_submission')
```

Comment:
We should embed some sort of UUID here (and maybe ensure it doesn't already exist at judge startup). Running multiple judge instances from the same machine is a supported configuration that'd break as-is (though prod would be fine because of Docker).

Reply:
Done.